src/v/cloud_storage_clients/test_client/abs_test_client_main.cc

```diff
@@ -0,0 +1,317 @@
+                        const auto result = cli
+                                              .put_object(
+                                                lcfg.container,
+                                                lcfg.blobs.front(),
+                                                payload_size,
+                                                std::move(payload),
+                                                {},
+                                                http::default_connect_timeout)
+                                              .get0();
+                        if (!result) {
+                            vlog(
+                              test_log.error,
+                              "PUT request failed: {}",
+                              result.error());
+                        }
+                    } else if (lcfg.delete_blob) {
+                        // Delete Blob
+                        vlog(test_log.info, "Deleting blob");
+                        const auto result = cli
+                                              .delete_object(
+                                                lcfg.container,
+                                                lcfg.blobs.front(),
+                                                http::default_connect_timeout)
+                                              .get();
+
+                        if (result) {
+                            vlog(test_log.info, "Delete Blob completed");
+                        } else {
+                            vlog(
+                              test_log.error,
+                              "Delete request failed: {}",
+                              result.error());
+                        }
+                    } else if (lcfg.get_metadata) {
+                        // Get Blob Metadata
+                        vlog(test_log.info, "Getting blob metadata");
+                        const auto result = cli
+                                              .head_object(
+                                                lcfg.container,
+                                                lcfg.blobs.front(),
+                                                http::default_connect_timeout)
+                                              .get();
+                        if (result) {
+                            vlog(
+                              test_log.info,
+                              "Get Blob Metadata completed: etag={}",
+                              result.value().etag);
+                        } else {
+                            vlog(
+                              test_log.error,
+                              "Get Blob Metadata request failed: {}",
+                              result.error());
+                        }
+                    }
+                })
+              .get();
+            client.stop().get();
+            vlog(test_log.info, "done");
+            ss::sleep(std::chrono::seconds(1)).get();
```

Comment:
Note for reviewer: the destructor for seastar's `tls::session` detaches a background thread to end the connection nicely. The sleep gives that thread a chance to join. See https://github.com/scylladb/seastar/blob/master/src/net/tls.cc#L1472.